### PR TITLE
Icchen/1826 condition of enabling catalog selection button

### LINKED ISSUE #1826: only enable catalog selection button when there is a layer of catalog overlay
**Describe the bug**
Once a catalog file is loaded, the catalog selection button in the image viewer is enabled and if we click it and click the image elsewhere, we will see the first source in the catalog table is selected. The catalog selection button should only be enabled when a catalog image overlay exists. 

**To Reproduce**
1. load an image and a catalog file
2. observe the state of the catalog selection button in the image viewer and it is enabled.
3. click the catalog selection button and click elsewhere in the image 
4. we will see the first source in the catalog table is highlighted in red.

**Expected behavior**
only enable the catalog selection button when a catalog overlay exists on top of the raster image.

**Screenshots or videos**
Screen Shot 2022-04-12 at 14 21 36

**Platform info (please complete the following information):**
 - OS: macOS Monterey
 - Browser chrome
 - Version 100.0.4896.60 (Official Build) (x86_64)

### PR DESCRIPTION
**Pull request by I-Chenn**
**Description**
Fix issue #1826. Now the condition of enabling "Catalog selection" button is:
1. **Catalog overlay is activated**
or
2. **There is at least one source is plotted on the viewer**

Result:
1. When "Catalog overlay" is activated, the "Catalog selection" button is still forbidden to click (note the mouse cursor):
buttonNotEnabled
(sorry for taking the picture with cellphone camera since print screen may let mouse cursor disappear)

2. When sources are plotted on the viewer, the  "Catalog selection" button is enabled
buttonEnabledWhenSourceIsPlotted

3. Nevertheless, this modification still doesn't fix the "**the first source in the catalog table is highlighted in red when click elsewhere on the image**" problem (even when the "Catalog selection" button is not enabled yet...):
image

**Checklist**
- [x] changelog updated / ~~no changelog update needed~~
- [x] e2e test passing / ~~added corresponding fix~~
- [x] ~~protobuf updated to the latest dev commit~~ / no protobuf update needed

### CODE REVIEW

**Comment by kswang1029** [comment]
ok for a formal PR

**Review by kswang1029 — APPROVED** [review]

File: src/components/ImageView/Toolbar/ToolbarComponent.tsx [hunk 1]
```diff
@@ -193,8 +193,20 @@ export class ToolbarComponent extends React.Component<ToolbarComponentProps> {
             </Menu>
         );
 
+        const baseFrame = this.props.frame;
```
Comment (YuHsuan-Hwang) [inline src/components/ImageView/Toolbar/ToolbarComponent.tsx]: We may need to consider spatially matched frames: `this.props.frame.spatialReference ?? this.props.frame`

File: src/components/ImageView/Toolbar/ToolbarComponent.tsx [hunk 2]
```diff
@@ -193,8 +193,20 @@ export class ToolbarComponent extends React.Component<ToolbarComponentProps> {
...
+        let numSourcesArray = [];
```
Comment (YuHsuan-Hwang) [inline src/components/ImageView/Toolbar/ToolbarComponent.tsx]: Perhaps we don't need to calculate the exact sum of source number. We can apply `Array.prototype.every()` to check if each element is zero.

**Review by YuHsuan-Hwang — CHANGES_REQUESTED** [review]

File: src/components/ImageView/Toolbar/ToolbarComponent.tsx [hunk 3]
```diff
@@ -193,8 +193,17 @@ export class ToolbarComponent extends React.Component<ToolbarComponentProps> {
             </Menu>
         );
 
+        const baseFrame = this.props.frame.spatialReference ?? this.props.frame;
+        let numSourcesArray = [];
+        appStore.catalogStore.visibleCatalogFiles.get(baseFrame).forEach(fileId => {
```
Comment (YuHsuan-Hwang) [inline src/components/ImageView/Toolbar/ToolbarComponent.tsx]: can be simplified by using `map` instead of `forEach`

**Review by YuHsuan-Hwang — CHANGES_REQUESTED** [review]

**Comment by I-Chenn** [comment]
@YuHsuan-Hwang I have modified the source code following your suggestion and squeeze the original source code into one line. Please let me know if you're not satisfied with that.

**Review by YuHsuan-Hwang — APPROVED** [review]

**Comment by YuHsuan-Hwang** [comment]
> Nevertheless, this modification still doesn't fix the "the first source in the catalog table is highlighted in red when click elsewhere on the image" problem (even when the "Catalog selection" button is not enabled yet...)

Filed a new issue for this problem.

**Pull request merged**